src/plugins/data/common/search/aggs/buckets/range_key.ts

```diff
@@ -19,20 +19,41 @@
 
 const id = Symbol('id');
 
+type Ranges = Array<{
+  from: string | number | undefined;
+  to: string | number | undefined;
+  label?: string;
+}>;
+
 export class RangeKey {
   [id]: string;
   gte: string | number;
   lt: string | number;
+  label?: string;
+
+  private static findCustomLabel(from: string | number, to: string | number, ranges: Ranges) {
```

Comment:
Making `ranges` optional here makes the callee code much shorter (the missing `ranges` is already handled in the body).